nist/int.go

```diff
@@ -299,10 +298,16 @@ func (i *Int) MarshalSize() int {
 func (i *Int) MarshalBinary() ([]byte, error) {
 	l := i.MarshalSize()
 	b := i.V.Bytes() // may be shorter than l
-	if ofs := l - len(b); ofs != 0 {
+	offset := l - len(b)
+
+	if i.Bo == LittleEndian {
+		return i.LittleEndian(l, l), nil
```

Comment:
Ooh, what is the reason for the LittleEndian() method to exist on this object.  Looks unnecessary and kinda hairy and gross to me.  Why not just delete it and instead:

1. Encode into a big-endian byte-slice, handling the offset correctly for big-endian as the code below does.
2. *after* that (before line 307/312 below), test for LittleEndian and conditionally to a util.Reverse on the encoded slice.

Wouldn't that be way simpler?  Am I missing anything?